core/services/functions/listener.go

```diff
@@ -297,13 +297,15 @@ func (l *functionsListener) HandleOffchainRequest(ctx context.Context, request *
 	if len(request.RequestId) != RequestIDLength {
 		return fmt.Errorf("HandleOffchainRequest: invalid request ID length %d", len(request.RequestId))
 	}
-	if len(request.SubscriptionOwner) != common.AddressLength || len(request.RequestInitiator) != common.AddressLength {
-		return fmt.Errorf("HandleOffchainRequest: SubscriptionOwner and RequestInitiator must be set to valid addresses")
+	if len(request.RequestInitiator) != common.AddressLength {
```

Comment:
Instead of losing a safety check could we instead fill the subscription owner with an empty address?

Reply:
This is only "lost" because I removed the SubscriptionOwner field altogether. But you have a point in your other comment that maybe it makes more sense to keep it for the future and just enforce zero for Heartbeats.